src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -185,39 +209,51 @@ private static void initializeMongoSqlTranslateLibrary() {
     private static boolean loadMongoSqlTranslateLibrary(String libraryPath) {
         try {
             System.load(libraryPath);
+            System.out.println("Sucessfully loaded " + libraryPath);
             return true;
         } catch (Throwable t) {
             return false;
         }
     }
 
-    public static CodecRegistry getCodecRegistry() {
-        return registry;
-    }
-
     // Resolves the potential paths where the MongoSQL Translate library are expected be located.
     private static String[] resolveLibraryPaths() throws Exception {
-        String libraryPath = getLibraryPath();
-
         // The `MONGOSQL_TRANSLATE_PATH` environment variable allows specifying an alternative library path.
         // This provides a backdoor mechanism to override the default library path of being colocated with the
         // driver library and load the MongoSQL Translate library from a different location.
         // Intended primarily for development and testing purposes.
         String envPath = System.getenv(MONGOSQL_TRANSLATE_PATH);
 
         List<String> paths = new ArrayList<>();
-        paths.add(libraryPath);
+
+        // Add the library path defined via  the `MONGOSQL_TRANSLATE_PATH` environment variable if it existed
         if (envPath != null && !envPath.isEmpty()) {
             paths.add(envPath);
         }
+
+        // Add the default library path
+        paths.add(getLibraryPath());
+
+        // Pick the first entry. If the `MONGOSQL_TRANSLATE_PATH` environment variable was present, then it's the one
+        // which will be picked, if not, then the default library will be loaded.
         return paths.toArray(new String[0]);
     }
 
     private static String getLibraryPath() throws Exception {
-        URL url = MongoDriver.class.getProtectionDomain().getCodeSource().getLocation();
-        Path driverPath = Paths.get(url.toURI());
-        Path driverDir = driverPath.getParent();
-        return driverDir.resolve(System.mapLibraryName(MONGOSQL_TRANSLATE_NAME)).toString();
+
+        String libName = System.mapLibraryName(MONGOSQL_TRANSLATE_NAME);
+        URL url =
```

Comment:
Now that it's checking for the resource here, it's possible that it isn't found and returns `null`.  Then the subsequent `url.toURI()` will get a null pointer exception.  Which I think still works because we catch the exception and return false for whether it is loaded.  It might be good to add a null check and a log message if null.

Reply:
We cannot log in MongoDriver.
I just added a mechanism to store the error found if loading the library failed and I am logging it at connection time.
I think that should also address this comment, but let me know if you had something else in mind.

Reply:
Yes, looks like it's addressed.  Thanks.